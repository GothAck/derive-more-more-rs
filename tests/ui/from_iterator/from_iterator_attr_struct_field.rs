use derive_more_more::FromIterator;

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct NamedMultiAttrStructField {
    #[from_iterator(usize)]
    inner: Vec<usize>,
    test: bool,
}

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct NamedSingleAttrStructField {
    #[from_iterator(usize)]
    inner: Vec<usize>,
}

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct UnnamedMultiAttrStructField(#[from_iterator(usize)] Vec<usize>, bool);

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct UnnamedSingleAttrStructField(#[from_iterator(usize)] Vec<usize>);

fn main() {}
