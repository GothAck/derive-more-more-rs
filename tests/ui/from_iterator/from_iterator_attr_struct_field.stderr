error: Expected `from_iterator` attribute on struct or one field
 --> tests/ui/from_iterator/from_iterator_attr_struct_field.rs:4:1
  |
4 | #[from_iterator(usize)]
  | ^

error: Expected `from_iterator` attribute on struct or one field
  --> tests/ui/from_iterator/from_iterator_attr_struct_field.rs:12:1
   |
12 | #[from_iterator(usize)]
   | ^

error: Expected `from_iterator` attribute on struct or one field
  --> tests/ui/from_iterator/from_iterator_attr_struct_field.rs:19:1
   |
19 | #[from_iterator(usize)]
   | ^

error: Expected `from_iterator` attribute on struct or one field
  --> tests/ui/from_iterator/from_iterator_attr_struct_field.rs:23:1
   |
23 | #[from_iterator(usize)]
   | ^
